lib/connect.js

```diff
@@ -44,6 +45,12 @@ function openSocket (host, port) {
   if (isIp(host)) {
     inner.open(host, port);
   } else {
+    // if URL contains "http://" or "https://" (etc), remove it.
+    if (host.indexOf('://') != -1) {
+      // http://www.example.com/superexample -> www.example.com
+      // UPDATE: use url.parse (much easier)
+      host = url.parse(host).host;
```

Comment:
`.hostname` could return null is URL is invalid, i guess we could throw an error in this case